doc/dev/autograph.rst

```diff
@@ -464,6 +463,24 @@ for instance by indexing a Python list with it. In that case, the list should be
 To understand different types of JAX tracing errors, please refer to the guide at: https://jax.readthedocs.io/en/latest/errors.html
 If you did not intend for the conversion to happen, you may safely ignore this warning.
 
+Something similar will also happen with list manipulations/operations:
+
+>>> @qjit(autograph=True)
+... def f():
+...     my_list = []
+...     for i in range(2):
+...         my_list.append(i)
+...     return my_list
+...
+>>> f()
+UserWarning: Tracing of an AutoGraph converted for loop failed with an exception:
+AutoGraphError: The variable 'my_list' was initialized with type <class 'list'>, which is not compatible with JAX. Typically, this is the case for
+non-numeric values. You may still use such a variable as a constant inside a loop, but it cannot be updated from one iteration to the next, 
+or accessed outside the loop scope if it was defined inside of it.
+...
+[Array(0, dtype=int64), Array(1, dtype=int64)]
+
+In this case, the code still executes, but AutoGraph is telling us that it fell back to executing the loop at compile time.
```

Comment:
Thank you Isaac! This is actually a good opportunity to document the currently supported pattern for accumulating results (with the caveat that the number and type of values must be known, and that they are homogenous):

```py
>>> @qjit(autograph=True)
... def f():
...     my_list = jnp.empty(2, dtype=int)
...     for i in range(2):
...         my_list[i] = i
...     return my_list
...
>>> f()
```

Reply:
Good suggestion! How about this:

```suggestion
In this case, the code still executes, but AutoGraph is telling us that it fell back to executing the loop at compile time. Instead, results can be accumulated by indexing into an empty, but fixed in shape/size, JAX array:

>>> @qjit(autograph=True)
... def f():
...     my_list = jnp.empty(2, dtype=int)
...     for i in range(2):
...         my_list[i] = i
...     return my_list
...
>>> f()
Array([0, 1], dtype=int64)

Here, AutoGraph is able to properly capture the results.
```

Reply:
Perfect, I made two wording changes to the suggestion but it looks great 👍